assets/src/dashboard/animations/animator/animation.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { GeneralAnimationPropTypes, KeyframesPropTypes } from './types';
+
+function Animation({ id, keyframes, ...options }) {
+  return (
+    <amp-animation id={id} layout="nodisplay">
```

Comment:
There's one annoying nuance. I don't believe amp-story currently allows amp-animation, although it really should :( It might take a bit more time than we have to enable. For that reason we may have to go with dual mode for CSS-vs-API animations. In the meantime, let's continue down the path you started and I'll first try to see if we can get the amp-animation enabled asap.

Reply:
I did do a test run and if we load the `amp-animation` tags above the `amp-story` tags, then it _will_ animate the elements within the `amp-story`.  I literally added in a simple animation into the `story.js` file to see if I could animate elements with a mask type of star and it worked!  That's why I kept the components separate, I believe we can load the `Animation` and `Animator` components before the `amp-story` and then use `WithAnimation` within the story and it should work.

Reply:
It will likely work in most cases, yes. But it currently will not validate and there are a couple of known bugs. I will try to resolve them asap. But not sure what's realistic here yet. Let's continue down this path and I'll bring this back up once it's clear.